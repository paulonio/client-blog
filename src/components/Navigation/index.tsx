import React, { FC } from 'react';
import Link from 'next/link';
import { useTranslation, Trans } from 'next-i18next';

import { parseString } from '@/utils/utils';
import { inter } from '@/styles/fonts';

import styles from './styled.module.scss';

export interface LinkType {
  href: string;
  text: string;
}

interface NavigationProps {
  links: LinkType[];
  onClick?: () => void;
}

const Navigation: FC<NavigationProps> = ({ links, onClick }) => {
  const { t } = useTranslation('header');

  return (
    <ul className={styles.navigation}>
      {links.map(({ href, text }) => (
        <li key={href} className={`${styles.navigation__link} ${inter.className}`}>
          <Trans i18nKey="nav">
            <Link href={href} onClick={onClick}>
              {t(parseString(text))}
            </Link>
          </Trans>
        </li>
      ))}
    </ul>
  );
};

export default Navigation;
