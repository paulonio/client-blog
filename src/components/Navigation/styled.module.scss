@import '../../styles/sass/constants';

.navigation {
  display: flex;
  align-items: center;
  gap: 24px;

  &__link {
    list-style: none;

    & a {
      font-size: 16px;
      line-height: 28px;
      font-weight: 400;
      color: $color-white;
      transition: color 0.3s;
      &:hover {
        color: $color-primary;
      }

      &.active {
        color: $color-primary;
      }
    }
  }

  @media screen and (max-width: 768px) {
    display: none;
  }
}