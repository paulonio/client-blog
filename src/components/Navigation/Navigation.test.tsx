import React from 'react';
import { render } from '@testing-library/react';
import Navigation from '.';
import type { LinkType } from '@/components/Navigation';

const TEST_LINKS: LinkType[] = [{ href: '/test', text: 'test' }];

describe('Navigation', () => {
  it('should render navigation', () => {
    const { container } = render(<Navigation links={TEST_LINKS} onClick={() => {}} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with empty array', () => {
    const { container } = render(<Navigation links={[]} onClick={() => {}} />);

    expect(container).toMatchSnapshot();
  });
});
