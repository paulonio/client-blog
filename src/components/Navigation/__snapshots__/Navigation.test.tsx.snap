// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render navigation 1`] = `
<div>
  <ul
    class="navigation"
  >
    <li
      class="navigation__link className"
    >
      <a
        href="/test"
      >
        test
      </a>
    </li>
  </ul>
</div>
`;

exports[`Navigation should render with empty array 1`] = `
<div>
  <ul
    class="navigation"
  />
</div>
`;
